browser/ui/views/omnibox/brave_omnibox_view_views.cc

```diff
@@ -121,6 +138,48 @@ void BraveOmniboxViewViews::ExecuteTextEditCommand(
 }
 #endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC)
 
+void BraveOmniboxViewViews::ExecuteCommand(int command_id, int event_flags) {
+  // Early return if |location_bar_view_| is null as it's used to get Browser
+  // instance pointer.
+  if (!location_bar_view_ || command_id != IDC_PASTE_AND_GO ||
+      !ShouldExecuteForPasteAndSearch()) {
+    return OmniboxViewViews::ExecuteCommand(command_id, event_flags);
+  }
+
+  // Do paste and search here instead of delegating to
+  // OmniboxEditModel::PasteAndGo(). In OmniboxEditModel, only normal
+  // profile's search provider is used because same AutocompleteClassifier is
+  // shared between normal and private profile.
+  constexpr size_t kMaxSelectionTextLength = 50;
+  const std::u16string clipboard_text = GetClipboardText();
+  CHECK(!clipboard_text.empty());
```

Comment:
Maybe we need some brief explanation why empty would check.

Reply:
Yes, correct. Slightly refactored to make it more clear.